routes/skills.py

```
@@ -1,260 +1,362 @@
+            db.session.commit()
+        except IntegrityError:
+            db.session.rollback()
+            flash('This variant already exists', 'error')
+            return redirect(url_for('skill.edit_skill', skill_id=skill_id))
+        
+        # Refresh the skill service cache since we added a variant
+        skill_service.refresh_cache()
+        
+        flash(f'Variant "{variant_name}" added successfully', 'success')
+        
     except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)